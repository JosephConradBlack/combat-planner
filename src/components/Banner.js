import React from 'react';
import styled from 'styled-components';
import { Segment, Header, Container } from 'semantic-ui-react';

import SiteMenu from './SiteMenu';

const StyledSegment = styled(Segment)`
  &&& {
    min-height: 400px;
    padding: 1em 0em;
  }
`;

const BigHeader = styled(Header)`
  &&& {
    font-size: ${props => props.theme.h1.fontSize};
    font-weight: ${props => props.theme.h1.fontWeight};
    letter-spacing: ${props => props.theme.h1.letterSpacing};
    margin-top: 100px;
  }
`;

const Banner = () => {
  return (
    <StyledSegment inverted vertical textAlign="center">
      <SiteMenu />
      <Container>
        <BigHeader inverted as="h1">
          Combat Planner
        </BigHeader>
      </Container>
    </StyledSegment>
  );
};

export default Banner;
