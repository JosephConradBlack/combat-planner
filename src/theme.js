export default {
  h1: {
    fontSize: '6em',
    fontWeight: 300,
    letterSpacing: '0.06em',
    subheader: {
      fontSize: '1em',
      fontWeight: 'lighter'
    }
  }
};
