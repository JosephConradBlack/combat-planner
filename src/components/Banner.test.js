import React from 'react';
import { create } from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';

import theme from '../theme';
import Banner from './Banner';

describe('Banner', () => {
  it('renders', () => {
    const component = create(
      <ThemeProvider theme={theme}>
        <Banner />
      </ThemeProvider>
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
