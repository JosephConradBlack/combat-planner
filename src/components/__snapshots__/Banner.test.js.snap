// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner renders 1`] = `
<div
  className="ui inverted vertical center aligned segment sc-bdVaJa lasqMc"
>
  <div
    className="ui large inverted pointing secondary menu"
  >
    <div
      className="ui container"
    >
      <div
        className="item"
        onClick={[Function]}
      >
        Combat Planner
      </div>
    </div>
  </div>
  <div
    className="ui container"
  >
    <h1
      className="ui inverted header sc-bwzfXH cKpNQC"
    >
      Combat Planner
    </h1>
  </div>
</div>
`;
